docs/user_manual.md

```diff
@@ -349,11 +349,55 @@ this way; it would be an error to apply `symbol(foo)` to another production in
 the module above. Additionally, `symbol(_)` with an argument may not co-occur
 with the `klabel(_)` attribute (see below).
 
+### `overload` attribute
+
+K supports _subsort overloading_[^maude-overload] on symbols, whereby a 
+constructor can have a more specific sort for certain arguments. For example,
+consider the following productions derived from a C-like language semantics:
+```k
+syntax Exp  ::= LVal
+              | Exp  "." Id
+syntax LVal ::= LVal "." Id
+```
+
+Here, it is useful for the result of the dot operator to be an `LVal` if the
+left-hand side is itself an `LVal`. However, there is an issue with the code
+as written: if `L()` is a term of sort `LVal`, then the program `L() . x` has a
+parsing ambiguity between the two productions for the dot operator. To resolve
+this, we can mark the productions as _overloads_:
+```k
+syntax Exp  ::= LVal
+              | Exp  "." Id [overload(_._)]
+syntax LVal ::= LVal "." Id [overload(_._)]
+```
+
+Now, the parser will select the _most specific_ overloaded production when it
+resolves ambiguities in `L() . x` (that is, `L() . x` parses to a term of sort
+`LVal`.
+
+Formally, the compiler organises productions into a partial order that defines
+the overload relation as follows. We say that `P` is a more specific overload
+of `Q` if:
+* `P` and `Q` have the same `overload(_)` attribute. Note that the argument
+  supplied has no semantic meaning other than as a key grouping productions 
+  together.
+* Let `S_P` be the sort of `P`, and `S_p1` etc. be the sorts of its arguments
+  (c.f. for `Q`). The tuple `(S_P, S_p1, ..., S_pN)` must be lexicographically
```

Comment:
This isn't a lexicographic ordering - overloading requires that **every** component of `P`'s tuple is a subsort of the corresponding component of `Q`'s tuple, but lexicographic ordering only requires this for the first component where the tuples disagree.

Formally, renaming `S_P` to `S_p0` for ease of notation, the overload ordering defines `P < Q` if
- `S_pi <: S_qi` for all `0 ≤ i ≤ N`
- `(S_P, S_p1, ..., S_pN) != (S_Q, S_q1, ..., S_qN)`

However, a lexicographic ordering would only require
- There exists some `0 ≤ k ≤ N` such that `S_pi = S_qi` for all `0 ≤ i < k` and `S_pk < S_qk`